app_dart/lib/src/model/appengine/branch.dart

```diff
@@ -0,0 +1,46 @@
+// Copyright 2021 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:gcloud/db.dart';
+import 'package:github/github.dart';
+
+@Kind(name: 'Branch', idType: IdType.String)
+class Branch extends Model<String> {
+  Branch({Key<String>? key, this.branch, this.defaultBranch, this.repository}) {
+    parentKey = key?.parent;
+    id = key?.id;
+  }
+
+  /// The branch name of the current branch
+  @StringProperty(propertyName: 'branch', required: true)
+  String? branch;
+
+  /// A serializable form of [RepositorySlug].
+  ///
+  /// This will be of the form `<org>/<repo>`. e.g. `flutter/flutter`.
+  @StringProperty(propertyName: 'FlutterRepositoryPath', required: true)
```

Comment:
`branch` and `repository` are stored in the key. Let's simplify and make these getters based on the key

Reply:
For sure, the branch class is now cut down to have only two fields